pkg/controller/serviceaccount/tokens_controller.go

```diff
@@ -256,10 +258,12 @@ func (e *TokensController) syncServiceAccount() {
 			klog.Errorf("error deleting serviceaccount tokens for %s/%s: %v", saInfo.namespace, saInfo.name, err)
 		}
 	default:
-		// ensure a token exists and is referenced by this service account
-		retry, err = e.ensureReferencedToken(sa)
-		if err != nil {
-			klog.Errorf("error synchronizing serviceaccount %s/%s: %v", saInfo.namespace, saInfo.name, err)
+		if !utilfeature.DefaultFeatureGate.Enabled(kubefeatures.LegacyServiceAccountTokenNoAutoGeneration) {
```

Comment:
I think it will be easier to test and understand if we make this an `Autogenerate bool` field in TokensControllerOptions and TokensController, and populate it at construction time from feature gates

Then you can make this the case condition (`case e.Autogenerate == true:`)